node/src/types/status_feed.rs

```diff
@@ -173,11 +164,12 @@ pub struct GetStatusResult {
 }
 
 impl GetStatusResult {
+    #[allow(deprecated)]
     pub(crate) fn new(status_feed: StatusFeed<NodeId>, api_version: ProtocolVersion) -> Self {
         GetStatusResult {
             api_version,
             chainspec_name: status_feed.chainspec_info.name,
-            starting_state_root_hash: status_feed.chainspec_info.starting_state_root_hash,
+            starting_state_root_hash: Some(Default::default()),
```

Comment:
As discussed, this is being deprecated.

We can change this to `None`

Reply:
I've documented that we should remove this in 2.0.0

Reply:
Fixed in 6fc17a4fd320d8e2cc910e2abc7dcb9dfa3dbc17

Reply:
> I see that you removed the `ChainspecLoader` entirely. Also, `ContractRuntimeRequest::CommitGenesis` and `ContractRuntimeRequest::Upgrade` events (and their handling logic) have been removed. Can you include the explanation of it in the description? Specifically, asnwer questions like:
> 
> * Why we're doing it?
> * Why we _can_ do it? Why it is OK to do that.

Nothing is constructing these effects, so it's dead code.

> * What is it being replaced with?

We're calling the genesis and upgrade methods directly in the participating reactor.

This was done in #1666 so we could construct a switch block immediately after genesis/upgade.

#1666 was done because fast-sync's security model won't permit joining a network after an emergency upgrade until it can find a switchblock.  So we'd be in a scenario where we'd have to wait two hours before anybody could get a trusted switch block in that case.

In #1666 @afck and needed to initialize the `EraSupervisor` properly.  The simplest solution was just to call the contract runtime directly in the participating reactor constructor.